service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ManageHearingsController.java

```diff
@@ -145,8 +147,18 @@ public CallbackResponse populateExistingDraftHearing(@RequestBody CallbackReques
         } else if (VACATE_HEARING == caseData.getHearingOption()) {
             UUID hearingBookingId = hearingsService.getSelectedHearingId(caseData);
 
-            caseDetails.getData().put(FUTURE_HEARING_LIST,
-                hearingsService.asDynamicList(caseData.getFutureAndTodayHearings(), hearingBookingId));
+            caseDetails.getData().put(VACATE_HEARING_LIST,
+                hearingsService.asDynamicList(caseData.getAllNonCancelledHearings(), hearingBookingId));
+
+            HearingBooking hearingBooking = hearingsService
+                .findHearingBooking(hearingBookingId, caseData.getHearingDetails())
+                .orElse(HearingBooking.builder().build());
+
+            caseDetails.getData().put("showVacatePastHearingWarning",
+                BooleanUtils.toString(hearingBooking.getEndDate().isBefore(now()), "Yes", "No"));
```

Comment:
Nice!
Maybe we could even have this tranformation in a class that could be re-used in the many other places in the code where a boolean becomes a string (I'm not saying you should make the transformation, but by having it in a separate class, it would be easier for other people to gradually work on this refactoring).